app/components/hiring_staff/vacancies_component.rb

```diff
@@ -49,4 +52,13 @@ def published
   def awaiting_feedback
     @organisation.vacancies.awaiting_feedback.order(@sort.column => @sort.order)
   end
+
+  def set_filters(vacancies, filters)
```

Comment:
this should be done with something like:

```  def set_filters(vacancies, filters)
    return vacancies if filters.none? || filters[:managed_organisations] == 'all'

    return vacancies.in_school_urns(filters[:managed_school_urns]) if filters[:managed_school_urns]&.any? && filters[:managed_organisations] != 'school_group'

    return vacancies.in_central_office if filters[:managed_organisations] == 'school_group' && filters[:managed_school_urns]&.none?

    vacancies.in_school_urns(filters[:managed_school_urns]).or(vacancies.joins(:school).in_central_office)
  end
```

Reply:
the or query causes problems because of the joins

Reply:
Both sides of the or query need to be the same structure (except for the where). We can't have a join for schools on the `in_central_office` scope because these vacancies don't have a school_id. We need to join on the `in_school_urns` scope because urn isn't a column on the database table. 

Approving the PR and merging for now, and this method will be re-implemented in due course. One potential option could be to manage organisation preferences by school_id instead of by urn, this way we don't need do a join at all.